DOC: move install related FAQ to install docs (#26628)
matplotlib/matplotlib: doc/devel/dependencies.rst

C++ compiler
------------

Matplotlib requires a C++ compiler that supports C++11. Typically, on Linux, you
will need ``gcc``, which should be installed using your distribution's package
manager; on macOS, you will need xcode; on Windows, you will need

> Visual Studio 2015 or later. Specifically:

Reviewer: These tabs/tables look much better than the bullet lists.

C++ compiler
------------

Matplotlib requires a C++ compiler that supports C++11, and each platform has a
development environment that must be installed before a compiler can be installed.

.. tab-set::

    .. tab-item:: Linux

        On some Linux systems, you can install a meta-build package. For example,
        on  Ubuntu ``apt install build-essential``

        Otherwise, use the system distribution's package manager to install

>         :ref:`gcc <compiler-table>`

Reviewer: ```suggestion
        :ref:`gcc <compiler-table>`.
```

>     .. tab-item:: OS/X

Reviewer: The official name is macOS
```suggestion
    .. tab-item:: macOS
```

>         Install `xcode <https://developer.apple.com/xcode/>`_ for Apple platform development

Reviewer: ```suggestion
        Install `Xcode <https://developer.apple.com/xcode/>`_ for Apple platform development.
```

    .. tab-item:: Windows

        Install `Visual Studio Build Tools <https://visualstudio.microsoft.com/downloads/?q=build+tools>`_

        Make sure "Desktop development with C++" is selected, and that the latest MSVC,
        "C++ CMake tools for Windows," and a Windows SDK compatible with your version
        of Windows are selected and installed. They should be selected by default under
        the "Optional" subheading, but are required to build Matplotlib from source.

        Alternatively, you can install a Linux-like environment such as `CygWin <https://www.cygwin.com/>`_
        or `Windows Subsystem for Linux <https://learn.microsoft.com/en-us/windows/wsl/install>`_.

We highly recommend that you install a compiler using your platform tool, i.e.,
Xcode, VS Code or Linux package manager. Choose **one** compiler from this list:

.. _compiler-table:

.. list-table::
   :widths: 20 20 20 40
   :header-rows: 1

   * - compiler
     - minimum version
     - platforms

>      - documenattion

Reviewer: maybe?
```suggestion
     - notes
```
but at least
```suggestion
     - documentation
```

>         Alternatively, you can install a Linux like environment like `CygWin <https://www.cygwin.com/>`_

Reviewer: ```suggestion
        Alternatively, you can install a Linux-like environment such as `CygWin <https://www.cygwin.com/>`_
```

> We highly recommend that you install a compiler using your platform tool, .i.e.
> xcode, vscode or linux package manager.

Reviewer: ```suggestion
We highly recommend that you install a compiler using your platform tool, i.e.,
Xcode, VS Code or Linux package manager.
```
